components/feature/prompts/src/main/java/mozilla/components/feature/prompts/dialog/LoginDialogFragment.kt

```diff
@@ -71,7 +71,17 @@ internal class LoginDialogFragment : PromptDialogFragment() {
     private var username by SafeArgString(KEY_LOGIN_USERNAME)
     private var password by SafeArgString(KEY_LOGIN_PASSWORD)
 
-    private var stateUpdate: Job? = null
+    private var validateStateUpdate: Job? = null
+
+    /**
+     * These are essential to the function of this fragment.
+     */
+    private val loginValidationDelegate by lazy {
```

Comment:
nit: do you need this to be `lazy`? that has overhead we may not need here. And below, replace `requireNotNull` with just `!!`?